server/service/hosts.go

```diff
@@ -735,12 +741,20 @@ func (svc *Service) MacadminsData(ctx context.Context, id uint) (*fleet.Macadmin
 		enrollmentStatus = "Enrolled (automated)"
 	}
 
-	data := &fleet.MacadminsData{
```

Comment:
If `err == fleet.IsNotFound(err)` you shouldn' try to read/use `enrolled, serverURL, installedFromDep` above.
```
	enrollmentStatus := "Unenrolled"
	if enrolled && !installedFromDep {
		enrollmentStatus = "Enrolled (manual)"
	} else if enrolled && installedFromDep {
		enrollmentStatus = "Enrolled (automated)"
	}
```